File: edge-plugin/src/main/scala/com/cloudentity/edge/plugin/RequestPluginService.java
package com.cloudentity.edge.plugin;

import com.cloudentity.edge.plugin.bus.request.*;
import com.cloudentity.tools.vertx.bus.VertxEndpoint;
import com.cloudentity.tools.vertx.tracing.TracingContext;
import io.vertx.core.Future;

public interface RequestPluginService extends ValidatePluginService, ConvertOpenApiService {
  @VertxEndpoint(address = ":request-plugin.apply")
  Future<ApplyResponse> applyPlugin(TracingContext ctx, ApplyRequest req);
}

File: pyron-plugin/src/main/scala/com/cloudentity/pyron/plugin/RequestPluginService.scala
package com.cloudentity.pyron.plugin

import com.cloudentity.pyron.plugin.bus.request._
import com.cloudentity.tools.vertx.bus.VertxEndpoint
import com.cloudentity.tools.vertx.tracing.TracingContext
import io.vertx.core.Future

trait RequestPluginService extends ValidatePluginService with ConvertOpenApiService {
  @VertxEndpoint(address = ":request-plugin.apply")
  def applyPlugin(ctx: TracingContext, req: ApplyRequest): Future[ApplyResponse]
}
